File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.HashSet;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        HashSet<String> phoneSet = new HashSet<>();
        
        for (String p : phone_book) {
            phoneSet.add(p);
        }
        
        for (String phone : phone_book) {
            for (int i = 1; i < phone.length(); i++) {
                if (phoneSet.contains(phone.substring(0, i))) {
                    return false;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    dct = {}
    for i in phone_book:
        dct[i] = 1
        
    for i in dct:
        check = ''
        for j in range(len(i)-1):
            check += i[j]
            if check in dct:
                return False
        
    return True